app/src/main/java/infrmr/newsapp/github/com/ifrmr/article/ArticleFragment.java

```diff
@@ -25,43 +77,87 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
-        // Option to resize large images
-        mWebView.getSettings().setLayoutAlgorithm(WebSettings.LayoutAlgorithm.SINGLE_COLUMN);
-        // Add the content to the view
-        mWebView.loadData(getHtmlContent(title, content), "text/html", null);
+        // Reference views
+        title = (TextView) layout.findViewById(R.id.articleLayoutTitle);
+        content = (TextView) layout.findViewById(R.id.articleLayoutContent);
+        updated = (TextView) layout.findViewById(R.id.articleViewUpdated);
+        loadingSpinner = (ProgressBar) layout.findViewById(R.id.loadingSpinner);
 
+        // Set up default view
+        title.setText(titleText);
+        content.setText(contentText);
         return layout;
     }
 
     /**
-     * Method for creating the article html code
-     *
-     * @param title - title of article
-     * @param content - article content
-     * @return - The completed string to be passed to WebView
+     * Cancel AsyncTask if in progress, as its no longer needed
      */
-    private String getHtmlContent(String title, String content) {
-
-        // Create StringBuilder object, used to build html string
-        StringBuilder htmlString = new StringBuilder();
-        // Used to get the updated time
-        Calendar rightNow = Calendar.getInstance();
-        DateFormat formatter = new SimpleDateFormat("MMM dd h:mmaa");
-
-        // Append string with title and article content
-        htmlString.append("<h3>").append(title).append("</h3>");
-        htmlString.append("<em>" + getResources().getString(R.string.updated) + " " +
-                formatter.format(rightNow.getTime()) + "</em>");
-        htmlString.append("<p>" + content + "</p><br><br>");
+    @Override
+    public void onStop() {
+        super.onStop();
+        if (parseTask != null) {
+            parseTask.cancel(true);
+        }
 
-        return htmlString.toString();
     }
 
-
+    /**
+     * AsyncTask for downloading article
+     */
+    class ParseURL extends AsyncTask<String, Void, String> {
+
+        @Override
+        protected String doInBackground(String... strings) {
+            StringBuilder buffer = new StringBuilder();
+            try {
+                Log.d(getClass().getSimpleName(), "JSoup - Connecting to [" + strings[0] + "]");
+                Document doc = Jsoup.connect(strings[0]).get();
+
+                int i = 0;
+                Elements paragraphs = doc.select("p");
+                for (Element paragraph : paragraphs) {
+                    if (paragraph.text().length() > paragraphToShortLength) {
+                        Log.i("JSOUP", "<p> - " + i + " - " + paragraph.text());
+                        buffer.append(paragraph.text()).append("\n\n");
+                    } else if (paragraph.text().contains("2015 Vox Media, Inc. All rights reserved")) {
+                        break;
+                    }
+
+                    i++;
+                }
+            } catch (Throwable t) {
```

Comment:
Avoid doing the "catch-all" - catch specific exceptions in their own blocks. Gives the opportunity to treat those conditions differently later on.